design/workload-scheduler.md

```diff
@@ -0,0 +1,252 @@
+
+```yaml
+apiVersion: compute.crossplane.io/v1alpha1
+kind: Workload
+metadata:
+  name: test-workload
+spec:
+  affinity:
+    clusterAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        clusterSelectorTerms:
+        - matchExpressions:
+          - key: crossplane.io/provider
+            operator: In
+            values:
+            - aws
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 1
+        preference:
+          matchExpressions:
+          - key: another-cluster-label-key
+            operator: In
+            values:
+            - another-cluster-label-value
+  resources:
+    - name: demo-mysql
+  targetNamespace: demo
+  targetDeployment:
+  targetService:	
+```
+	
+**TODO**: Cluster Affinity needs to be defined further.
+
+#### Inter-Workload Affinity and Anti-affinity
+Inter-workload affinity and anti-affinity allow you to constrain which clusters your workload is eligible to be scheduled based on labels on workloads that are already running on the cluster rather than based on labels on clusters. The rules are of the form “this workload should (or, in the case of anti-affinity, should not) run in an X if that X is already running one or more workloads that meet rule Y”. Conceptually X is a topology domain like cloud provider, cloud provider zone, cloud provider region, etc. You express it using a topologyKey which is the key for the cluster label that the system uses to denote such a topology domain, e.g. see the label keys listed above in the section ClusterSelector: built-in cluster labels. 
+
+In practical terms, Inter-Workload Affinity and Anti-affinity is important when deploying interdependent workloads, which must target the same cluster. For example, the front-end application must be deployed into the same Kubernetes cluster which is targeted by the back-end service.
+
+```yaml
+apiVersion: compute.crossplane.io/v1alpha1
+kind: Workload
+metadata:
+  name: front-end
+spec:
+  affinity:
+   workloadAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+          - key: app
+            operator: In
+            values:
+            - back-end
+        topologyKey: crossplane.io/kubernetes
+    workloadAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        workloadAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: app
+              operator: In
+              values:
+              - frontend
+          topologyKey: crossplane.io/kubernetes
+  resources:
+    - name: demo-mysql
+  targetNamespace: demo
+  targetDeployment:
+  targetService:	
+```
+
+**TODO**: Workload Affinity and Anti-affinity need to be defined further
+
+### Kubernetes Cluster Priorities
```

Comment:
Yes, your understanding is correct. I also agree the description is confusing. Updated. @negz PTAL

Reply:
The reference to the main Kubernetes scheduler implementation clears things up a little. I think it would be illustrative to show a worked example of a scheduling process, as I found it hard to understand at first why we'd need to prioritise in addition to applying predicated for our use cases.